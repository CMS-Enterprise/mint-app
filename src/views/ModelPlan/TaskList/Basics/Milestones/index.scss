.milestone-form {
  .fieldGroup__wrapper {
    display: flex;
    grid-gap: units(3);
    align-items: flex-end;
    flex-wrap: wrap;

    & > div {
      flex: 1;
    }
  }
}
