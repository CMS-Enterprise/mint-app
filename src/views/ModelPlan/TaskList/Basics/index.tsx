import React, { Fragment, useRef, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Link, Route, Switch, useHistory, useParams } from 'react-router-dom';
import { useMutation, useQuery } from '@apollo/client';
import {
  Breadcrumb,
  BreadcrumbBar,
  BreadcrumbLink,
  Button,
  Dropdown,
  Grid,
  GridContainer,
  IconArrowBack,
  Label,
  TextInput
} from '@trussworks/react-uswds';
import { Field, FieldArray, Form, Formik, FormikProps } from 'formik';

import AskAQuestion from 'components/AskAQuestion';
import MainContent from 'components/MainContent';
import PageHeading from 'components/PageHeading';
import PageNumber from 'components/PageNumber';
import Alert from 'components/shared/Alert';
import AutoSave from 'components/shared/AutoSave';
import CheckboxField from 'components/shared/CheckboxField';
import { ErrorAlert, ErrorAlertMessage } from 'components/shared/ErrorAlert';
import FieldErrorMsg from 'components/shared/FieldErrorMsg';
import FieldGroup from 'components/shared/FieldGroup';
import GetModelPlanInfo from 'queries/Basics/GetModelPlanInfo';
import {
  GetModelPlanInfo as GetModelPlanInfoType,
  GetModelPlanInfo_modelPlan as ModelPlanInfoFormType,
  GetModelPlanInfoVariables
} from 'queries/Basics/types/GetModelPlanInfo';
import { UpdatePlanBasicsVariables } from 'queries/Basics/types/UpdatePlanBasics';
import UpdatePlanBasics from 'queries/Basics/UpdatePlanBasics';
import { UpdateModelPlanVariables } from 'queries/types/UpdateModelPlan';
import UpdateModelPlan from 'queries/UpdateModelPlan';
import {
  CMMIGroup,
  CMSCenter,
  ModelCategory
} from 'types/graphql-global-types';
import flattenErrors from 'utils/flattenErrors';
import {
  translateCmmiGroups,
  translateCmsCenter,
  translateModelCategory
} from 'utils/modelPlan';
import planBasicsSchema from 'validations/planBasics';
import { NotFoundPartial } from 'views/NotFound';

import Milestones from './Milestones';
import Overview from './Overview';

const BasicsContent = () => {
  const { t } = useTranslation('basics');
  const { t: h } = useTranslation('draftModelPlan');
  const { modelID } = useParams<{ modelID: string }>();

  const formikRef = useRef<FormikProps<ModelPlanInfoFormType>>(null);
  const history = useHistory();
  const [areCmmiGroupsShown, setAreCmmiGroupsShown] = useState(false);
  const [showOther, setShowOther] = useState(false);

  const { data, loading, error } = useQuery<
    GetModelPlanInfoType,
    GetModelPlanInfoVariables
  >(GetModelPlanInfo, {
    variables: {
      id: modelID
    }
  });

  const { id, modelName, basics } = data?.modelPlan || {};

  const { modelCategory, cmsCenters, cmmiGroups, cmsOther } = basics || {};

  const [update] = useMutation<UpdateModelPlanVariables>(UpdateModelPlan);
  const [updateTwo] = useMutation<UpdatePlanBasicsVariables>(UpdatePlanBasics);

  const handleFormSubmit = (
    formikValues: ModelPlanInfoFormType,
    redirect?: 'next' | 'back'
  ) => {
    if (!formikValues.modelName) {
      formikRef?.current?.setFieldError('modelName', 'Enter the Model name');
      return;
    }
    const { id: updateId, __typename, ...changeValues } = formikValues;
    console.log(changeValues);
    //   try {
    //     const response = await update({
    //       variables: {
    //         id: updateId,
    //         changes: {
    //           modelName
    //         }
    //       }
    //     });
    //     if (!response?.errors) {
    //       if (redirect === 'next') {
    //         history.push(`/models/${modelID}/task-list/basics/overview`);
    //       } else if (redirect === 'back') {
    //         history.push(`/models/${modelID}/task-list/`);
    //       }
    //     }
    //   } catch (errors) {
    //     formikRef?.current?.setErrors(errors);
    //   }
    // };

    update({
      variables: {
        id: updateId,
        changes: {
          changeValues
        }
      }
    })
      .then(response => {
        if (!response?.errors) {
          if (redirect === 'next') {
            history.push(`/models/${modelID}/task-list/basics/overview`);
          } else if (redirect === 'back') {
            history.push(`/models/${modelID}/task-list/`);
          }
        }
      })
      .catch(errors => {
        formikRef?.current?.setErrors(errors);
      });
  };

  const initialValues: ModelPlanInfoFormType = {
    __typename: 'ModelPlan',
    id: id ?? '',
    modelName: modelName ?? '',
    basics: {
      __typename: 'PlanBasics',
      modelCategory: modelCategory ?? null,
      cmsCenters: cmsCenters ?? [],
      cmmiGroups: cmmiGroups ?? [],
      cmsOther: cmsOther ?? ''
    }
  };

  // 4 options
  // 1. Basics (name, category, CMS Component without CMMI and Other)
  // 2. Basics + cmmi group
  // 3. Basics + other group
  // 4. Basics + Cmmi + Other
  let validationSchema;
  if (areCmmiGroupsShown && showOther) {
    validationSchema = planBasicsSchema.pageOneSchemaWithOtherAndCmmi;
  } else if (areCmmiGroupsShown) {
    validationSchema = planBasicsSchema.pageOneSchemaWithCmmiGroups;
  } else if (showOther) {
    validationSchema = planBasicsSchema.pageOneSchemaWithOther;
  } else {
    validationSchema = planBasicsSchema.pageOneSchema;
  }

  if ((!loading && error) || (!loading && !data?.modelPlan)) {
    return <NotFoundPartial />;
  }

  return (
    <>
      <BreadcrumbBar variant="wrap">
        <Breadcrumb>
          <BreadcrumbLink asCustom={Link} to="/">
            <span>{h('home')}</span>
          </BreadcrumbLink>
        </Breadcrumb>
        <Breadcrumb>
          <BreadcrumbLink asCustom={Link} to={`/models/${modelID}/task-list/`}>
            <span>{h('tasklistBreadcrumb')}</span>
          </BreadcrumbLink>
        </Breadcrumb>
        <Breadcrumb current>{t('breadcrumb')}</Breadcrumb>
      </BreadcrumbBar>
      <PageHeading className="margin-top-4">{t('heading')}</PageHeading>

      <AskAQuestion modelID={modelID} />

      <Alert
        type="info"
        slim
        data-testid="mandatory-fields-alert"
        className="margin-bottom-4"
      >
        <span className="mandatory-fields-alert__text">
          {h('mandatoryFields')}
        </span>
      </Alert>

      <Formik
        initialValues={initialValues}
        onSubmit={values => {
          handleFormSubmit(values, 'next');
        }}
        enableReinitialize
        validationSchema={validationSchema}
        validateOnBlur={false}
        validateOnChange={false}
        validateOnMount={false}
        innerRef={formikRef}
      >
        {(formikProps: FormikProps<ModelPlanInfoFormType>) => {
          const {
            dirty,
            errors,
            handleSubmit,
            setErrors,
            setFieldValue,
            isValid,
            values
          } = formikProps;
          const flatErrors = flattenErrors(errors);
          return (
            <>
              {Object.keys(errors).length > 0 && (
                <ErrorAlert
                  testId="formik-validation-errors"
                  classNames="margin-top-3"
                  heading={h('checkAndFix')}
                >
                  {Object.keys(flatErrors).map(key => {
                    return (
                      <ErrorAlertMessage
                        key={`Error.${key}`}
                        errorKey={key}
                        message={flatErrors[key]}
                      />
                    );
                  })}
                </ErrorAlert>
              )}
              <Form
                className="tablet:grid-col-6 margin-top-6"
                onSubmit={e => {
                  handleSubmit(e);
                  window.scrollTo(0, 0);
                }}
              >
                <FieldGroup
                  scrollElement="modelName"
                  error={!!flatErrors.modelName}
                  className="margin-top-4"
                >
                  <Label htmlFor="plan-basics-model-name">
                    {t('modelName')}
                  </Label>
                  <FieldErrorMsg>{flatErrors.modelName}</FieldErrorMsg>
                  <Field
                    as={TextInput}
                    error={!!flatErrors.modelName}
                    id="plan-basics-model-name"
                    maxLength={50}
                    name="modelName"
                  />
                </FieldGroup>

                <FieldGroup
                  scrollElement="modelCategory"
                  error={!!flatErrors.modelCategory}
                  className="margin-top-4"
                >
                  <Label htmlFor="plan-basics-model-category">
                    {t('modelCategory')}
                  </Label>
                  <FieldErrorMsg>{flatErrors.modelCategory}</FieldErrorMsg>
                  <Field
                    as={Dropdown}
                    id="plan-basics-model-category"
                    name="basics.modelCategory"
                    value={values.basics.modelCategory || ''}
                    onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
                      setFieldValue('basics.modelCategory', e.target.value);
                    }}
                  >
                    <option key="default-select" disabled value="">
                      {`-${h('select')}-`}
                    </option>
                    {Object.keys(ModelCategory).map(category => {
                      return (
                        <option key={category} value={category || ''}>
                          {translateModelCategory(category)}
                        </option>
                      );
                    })}
                  </Field>
                </FieldGroup>

                <FieldGroup
                  scrollElement="cmsCenters"
                  error={!!flatErrors.cmsCenters}
                  className="margin-top-4"
                >
                  <FieldArray
                    name="cmsCenters"
                    render={arrayHelpers => (
                      <>
                        <legend className="usa-label">
                          {t('cmsComponent')}
                        </legend>
                        <FieldErrorMsg>{flatErrors.cmsCenters}</FieldErrorMsg>

                        {Object.keys(CMSCenter).map(center => {
                          return (
                            <Fragment key={center}>
                              <Field
                                as={CheckboxField}
                                id={`new-plan-cmsCenters-${center}`}
                                name="basics.cmsCenters"
                                label={translateCmsCenter(center)}
                                value={center}
                                checked={values.basics.cmsCenters.includes(
                                  center as CMSCenter
                                )}
                              />
                            </Fragment>
                          );
                        })}

                        {values.basics.cmsCenters.includes(CMSCenter.OTHER) && (
                          <FieldGroup
                            className="margin-top-4"
                            error={!!flatErrors.cmsOther}
                          >
                            <Label htmlFor="plan-basics-cmsCategory--Other">
                              {h('pleaseSpecify')}
                            </Label>
                            <FieldErrorMsg>{flatErrors.cmsOther}</FieldErrorMsg>
                            <Field
                              as={TextInput}
                              id="plan-basics-cmsCategory--Other"
                              maxLength={50}
                              name="basics.cmsOther"
                            />
                          </FieldGroup>
                        )}
                      </>
                    )}
                  />
                </FieldGroup>
                {values.basics.cmsCenters.includes(CMSCenter.CMMI) && (
                  <FieldGroup
                    error={!!flatErrors.cmmiGroups}
                    className="margin-top-4"
                  >
                    <Label htmlFor="basics.cmmiGroups">{t('cmmiGroup')}</Label>
                    <FieldErrorMsg>{flatErrors.cmmiGroups}</FieldErrorMsg>
                    {Object.keys(CMMIGroup).map(group => {
                      return (
                        <Fragment key={group}>
                          <Field
                            as={CheckboxField}
                            id={`new-plan-cmmiGroup-${group}`}
                            name="basics.cmmiGroups"
                            label={translateCmmiGroups(group)}
                            value={group}
                            checked={values.basics.cmmiGroups.includes(
                              group as CMMIGroup
                            )}
                          />
                        </Fragment>
                      );
                    })}
                  </FieldGroup>
                )}

                <div className="margin-top-6 margin-bottom-3">
                  <Button
                    type="submit"
                    disabled={!(dirty || isValid)}
                    onClick={() => setErrors({})}
                  >
                    {h('next')}
                  </Button>
                </div>
                <Button
                  type="button"
                  className="usa-button usa-button--unstyled"
                  onClick={() => handleFormSubmit(values, 'back')}
                >
                  <IconArrowBack className="margin-right-1" aria-hidden />
                  {h('saveAndReturn')}
                </Button>
              </Form>
              <AutoSave
                values={values}
                onSave={() => {
                  if (formikRef.current!.values.modelName)
                    handleFormSubmit(formikRef.current!.values);
                }}
                debounceDelay={3000}
              />
            </>
          );
        }}
      </Formik>
      <PageNumber currentPage={1} totalPages={3} className="margin-bottom-10" />
    </>
  );
};

export const Basics = () => {
  return (
    <MainContent data-testid="model-plan-basics">
      <GridContainer>
        <Grid desktop={{ col: 12 }}>
          <Switch>
            <Route
              path="/models/:modelID/task-list/basics"
              exact
              render={() => <BasicsContent />}
            />
            <Route
              path="/models/:modelID/task-list/basics/overview"
              exact
              render={() => <Overview />}
            />
            <Route
              path="/models/:modelID/task-list/basics/milestones"
              exact
              render={() => <Milestones />}
            />
            <Route path="*" render={() => <NotFoundPartial />} />
          </Switch>
        </Grid>
      </GridContainer>
    </MainContent>
  );
};

export default Basics;
