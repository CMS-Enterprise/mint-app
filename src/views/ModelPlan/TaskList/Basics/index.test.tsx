import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render, screen, waitFor } from '@testing-library/react';

import GetModelPlan from 'queries/GetModelPlan';
import {
  CMMIGroup,
  CMSCenter,
  ModelCategory,
  ModelStatus
} from 'types/graphql-global-types';

import Basics from './index';

const mocks = [
  {
    request: {
      query: GetModelPlan,
      variables: { id: 'f11eb129-2c80-4080-9440-439cbe1a286f' }
    },
    result: {
      data: {
        modelPlan: {
          modelName: 'My excellent plan that I just initiated',
          __typename: 'ModelPlan',
          id: 'f11eb129-2c80-4080-9440-439cbe1a286f',
          status: ModelStatus.PLAN_DRAFT,
          modelCategory: ModelCategory.PRIMARY_CARE_TRANSFORMATION,
          cmmiGroups: [
            CMMIGroup.STATE_INNOVATIONS_GROUP,
            CMMIGroup.POLICY_AND_PROGRAMS_GROUP
          ],
          cmsCenters: [CMSCenter.CENTER_FOR_MEDICARE, CMSCenter.OTHER],
          cmsOther: 'The Center for Awesomeness ',
          archived: false,
          discussions: [],
          generalCharacteristics: [],
          participantsAndProviders: [],
          opsEvalAndLearning: [],
          basics: [],
          documents: [],
          milestones: [],
          modifiedDts: ''
        }
      }
    }
  }
];

describe('Model Plan Documents page', () => {
  it('renders without errors', async () => {
    render(
      <MemoryRouter
        initialEntries={[
          '/models/f11eb129-2c80-4080-9440-439cbe1a286f/task-list/basics'
        ]}
      >
        <MockedProvider>
          <Route path="/models/:modelID/task-list/basics">
            <Basics />
          </Route>
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByTestId('model-plan-basics')).toBeInTheDocument();
    });
  });
  it('matches snapshot', async () => {
    const { asFragment } = render(
      <MemoryRouter
        initialEntries={[
          '/models/f11eb129-2c80-4080-9440-439cbe1a286f/task-list/basics'
        ]}
      >
        <MockedProvider mocks={mocks} addTypename={false}>
          <Route path="/models/:modelID/task-list/basics">
            <Basics />
          </Route>
        </MockedProvider>
      </MemoryRouter>
    );
    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
