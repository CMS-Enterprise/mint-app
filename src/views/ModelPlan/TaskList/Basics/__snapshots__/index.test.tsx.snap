// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Model Plan Documents page matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="mint-main-content"
    data-testid="model-plan-basics"
    id="main-content"
    role="main"
    tabindex="-1"
  >
    <div
      class="grid-container"
      data-testid="gridContainer"
    >
      <div
        class="desktop:grid-col-12"
        data-testid="grid"
      >
        <nav
          aria-label="Breadcrumbs"
          class="usa-breadcrumb usa-breadcrumb--wrap"
        >
          <ol
            class="usa-breadcrumb__list"
          >
            <li
              class="usa-breadcrumb__list-item"
            >
              <a
                class="usa-breadcrumb__link"
                href="/"
              >
                <span>
                  Home
                </span>
              </a>
            </li>
            <li
              class="usa-breadcrumb__list-item"
            >
              <a
                class="usa-breadcrumb__link"
                href="/models/f11eb129-2c80-4080-9440-439cbe1a286f/task-list/"
              >
                <span>
                  Model Plan task list
                </span>
              </a>
            </li>
            <li
              aria-current="page"
              class="usa-breadcrumb__list-item usa-current"
            >
              Model basics
            </li>
          </ol>
        </nav>
        <h1
          aria-live="polite"
          class="mint-h1 margin-top-4"
          tabindex="-1"
        >
          Model basics
        </h1>
        <div
          class="padding-2 bg-primary-lighter"
        >
          <div
            class="display-flex"
            data-testid="ask-a-question"
          >
            <svg
              class="usa-icon text-primary margin-right-1"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 9h-2V5h2v6zm0 4h-2v-2h2v2z"
              />
            </svg>
            <button
              class="usa-button usa-button--unstyled"
              data-testid="ask-a-question-button"
              type="button"
            >
              Ask a question
            </button>
          </div>
        </div>
        <p
          class="margin-bottom-0 margin-top-6"
        >
          All fields marked with 
          <span
            aria-label="(Field required)"
            class="text-red margin-x-05"
          >
            *
          </span>
           are required.
        </p>
        <div
          class="grid-container padding-x-0"
          data-testid="gridContainer"
        >
          <div
            class="grid-row grid-gap"
            data-testid="grid"
          >
            <div
              class="desktop:grid-col-6"
              data-testid="grid"
            >
              <form
                action="#"
                class="margin-top-4"
              >
                <div
                  class="usa-form-group margin-top-4"
                  data-scroll="modelName"
                >
                  <label
                    class="usa-label"
                    data-testid="label"
                    for="plan-basics-model-name"
                  >
                    Model name
                    <span
                      aria-label="(Field required)"
                      class="text-red margin-x-05"
                    >
                      *
                    </span>
                  </label>
                  <input
                    class="usa-input"
                    data-testid="textInput"
                    id="plan-basics-model-name"
                    maxlength="50"
                    name="modelName"
                    value="My excellent plan that I just initiated"
                  />
                </div>
                <div
                  class="usa-form-group margin-top-4"
                  data-scroll="abbreviation"
                >
                  <label
                    class="usa-label"
                    data-testid="label"
                    for="plan-basics-model-name"
                  >
                    Short name
                  </label>
                  <span
                    class="usa-hint display-block text-normal margin-top-1"
                  >
                    The abbreviation, acronym, or other common name used for the model.
                  </span>
                  <input
                    class="usa-input"
                    data-testid="textInput"
                    id="plan-basics-abbreviation"
                    maxlength="50"
                    name="abbreviation"
                    value="MEP"
                  />
                </div>
                <div
                  class="bg-base-lightest padding-2 margin-top-4"
                >
                  <label
                    class="usa-label margin-top-0"
                    data-testid="label"
                    for="plan-basics-demo-code"
                  >
                    Other identifiers
                  </label>
                  <p
                    class="line-height-mono-4"
                  >
                    These are created in 
                    <a
                      aria-label="Open AMS in a new tab"
                      class="usa-link usa-link--external"
                      href="https://ams.cmmi.cms.gov"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      CMMI Analysis & Management System (AMS).
                    </a>
                     Skip these fields until your model has been added to AMS. Not all models will have demo codes.
                  </p>
                  <div
                    class="grid-row grid-gap"
                    data-testid="grid"
                  >
                    <div
                      class="desktop:grid-col-6"
                      data-testid="grid"
                    >
                      <div
                        class="usa-form-group margin-top-0"
                        data-scroll="basics.amsModelID"
                      >
                        <label
                          class="usa-label"
                          data-testid="label"
                          for="plan-basics-ams-model-id"
                        >
                          Model ID
                        </label>
                        <input
                          class="usa-input"
                          data-testid="textInput"
                          id="plan-basics-ams-model-id"
                          maxlength="50"
                          name="basics.amsModelID"
                          value="2414213"
                        />
                      </div>
                    </div>
                    <div
                      class="desktop:grid-col-6"
                      data-testid="grid"
                    >
                      <div
                        class="usa-form-group margin-top-0"
                        data-scroll="basics.demoCode"
                      >
                        <label
                          class="usa-label"
                          data-testid="label"
                          for="plan-basics-demo-code"
                        >
                          Demo code(s)
                        </label>
                        <input
                          class="usa-input"
                          data-testid="textInput"
                          id="plan-basics-demo-code"
                          maxlength="50"
                          name="basics.demoCode"
                          value="123"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="usa-form-group margin-top-4"
                  data-scroll="modelCategory"
                >
                  <label
                    class="usa-label"
                    data-testid="label"
                    for="plan-basics-model-category"
                  >
                    Model category
                    <span
                      aria-label="(Field required)"
                      class="text-red margin-x-05"
                    >
                      *
                    </span>
                  </label>
                  <select
                    class="usa-select"
                    data-testid="dropdown"
                    id="plan-basics-model-category"
                    name="basics.modelCategory"
                  >
                    <option
                      disabled=""
                      value=""
                    >
                      -Select-
                    </option>
                    <option
                      value="ACCOUNTABLE_CARE"
                    >
                      Accountable Care
                    </option>
                    <option
                      value="DEMONSTRATION"
                    >
                      Demonstration
                    </option>
                    <option
                      value="EPISODE_BASED_PAYMENT_INITIATIVES"
                    >
                      Episode-based Payment Initiatives
                    </option>
                    <option
                      value="INIT_ACCEL_DEV_AND_TEST"
                    >
                      Initiatives to Accelerate the Development and Testing of New Payment and Service Delivery Models
                    </option>
                    <option
                      value="INIT_MEDICAID_CHIP_POP"
                    >
                      Initiatives Focused on the Medicaid and CHIP Population
                    </option>
                    <option
                      value="INIT_SPEED_ADOPT_BEST_PRACTICE"
                    >
                      Initiatives to Speed the Adoption of Best Practices
                    </option>
                    <option
                      value="INIT__MEDICARE_MEDICAID_ENROLLEES"
                    >
                      Initiatives Focused on the Medicare and Medicaid Enrollees
                    </option>
                    <option
                      value="PRIMARY_CARE_TRANSFORMATION"
                    >
                      Primary Care Transformation
                    </option>
                    <option
                      value="UNKNOWN"
                    >
                      Unknown
                    </option>
                  </select>
                </div>
                <div
                  class="usa-form-group margin-top-4"
                  data-scroll="cmsCenters"
                >
                  <fieldset
                    class="usa-fieldset"
                    data-testid="fieldset"
                  >
                    <legend
                      class="usa-legend"
                    >
                      CMS component
                    </legend>
                    <label
                      class="usa-label"
                      data-testid="label"
                      for="plan-basics-cmsCenters"
                    >
                      CMS component
                      <span
                        aria-label="(Field required)"
                        class="text-red margin-x-05"
                      >
                        *
                      </span>
                    </label>
                    <div
                      class="mint-checkbox usa-checkbox"
                    >
                      <input
                        class="usa-checkbox__input"
                        id="new-plan-cmsCenters-CMMI"
                        name="basics.cmsCenters"
                        type="checkbox"
                        value="CMMI"
                      />
                      <label
                        class="usa-checkbox__label"
                        for="new-plan-cmsCenters-CMMI"
                      >
                        CMMI
                      </label>
                      <p
                        class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs"
                      />
                    </div>
                    <div
                      class="mint-checkbox usa-checkbox"
                    >
                      <input
                        class="usa-checkbox__input"
                        id="new-plan-cmsCenters-CENTER_FOR_MEDICARE"
                        name="basics.cmsCenters"
                        type="checkbox"
                        value="CENTER_FOR_MEDICARE"
                      />
                      <label
                        class="usa-checkbox__label"
                        for="new-plan-cmsCenters-CENTER_FOR_MEDICARE"
                      >
                        Center for Medicare (CM)
                      </label>
                      <p
                        class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs"
                      />
                    </div>
                    <div
                      class="mint-checkbox usa-checkbox"
                    >
                      <input
                        class="usa-checkbox__input"
                        id="new-plan-cmsCenters-FEDERAL_COORDINATED_HEALTH_CARE_OFFICE"
                        name="basics.cmsCenters"
                        type="checkbox"
                        value="FEDERAL_COORDINATED_HEALTH_CARE_OFFICE"
                      />
                      <label
                        class="usa-checkbox__label"
                        for="new-plan-cmsCenters-FEDERAL_COORDINATED_HEALTH_CARE_OFFICE"
                      >
                        Federal Coordinated Health Care
                      </label>
                      <p
                        class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs"
                      />
                    </div>
                    <div
                      class="mint-checkbox usa-checkbox"
                    >
                      <input
                        class="usa-checkbox__input"
                        id="new-plan-cmsCenters-CENTER_FOR_CLINICAL_STANDARDS_AND_QUALITY"
                        name="basics.cmsCenters"
                        type="checkbox"
                        value="CENTER_FOR_CLINICAL_STANDARDS_AND_QUALITY"
                      />
                      <label
                        class="usa-checkbox__label"
                        for="new-plan-cmsCenters-CENTER_FOR_CLINICAL_STANDARDS_AND_QUALITY"
                      >
                        Center for Clinical Standards and Quality (CCSQ)
                      </label>
                      <p
                        class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs"
                      />
                    </div>
                    <div
                      class="mint-checkbox usa-checkbox"
                    >
                      <input
                        class="usa-checkbox__input"
                        id="new-plan-cmsCenters-CENTER_FOR_PROGRAM_INTEGRITY"
                        name="basics.cmsCenters"
                        type="checkbox"
                        value="CENTER_FOR_PROGRAM_INTEGRITY"
                      />
                      <label
                        class="usa-checkbox__label"
                        for="new-plan-cmsCenters-CENTER_FOR_PROGRAM_INTEGRITY"
                      >
                        Center for Program Integrity (CPI)
                      </label>
                      <p
                        class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs"
                      />
                    </div>
                    <div
                      class="mint-checkbox usa-checkbox"
                    >
                      <input
                        class="usa-checkbox__input"
                        id="new-plan-cmsCenters-OTHER"
                        name="basics.cmsCenters"
                        type="checkbox"
                        value="OTHER"
                      />
                      <label
                        class="usa-checkbox__label"
                        for="new-plan-cmsCenters-OTHER"
                      >
                        Other
                      </label>
                      <p
                        class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs"
                      />
                    </div>
                    <div
                      class="usa-form-group margin-top-4"
                    >
                      <label
                        class="usa-label"
                        data-testid="label"
                        for="plan-basics-cmsCategory--Other"
                      >
                        Please specify
                      </label>
                      <textarea
                        aria-labelledby="label-plan-basics-cmsCategory--Other hint-plan-basics-cmsCategory--Other"
                        class="mint-textarea usa-textarea"
                        id="plan-basics-cmsCategory--Other"
                        maxlength="5000"
                        name="basics.cmsOther"
                      >
                        The Center for Awesomeness 
                      </textarea>
                    </div>
                  </fieldset>
                </div>
                <div
                  class="usa-form-group margin-top-4"
                >
                  <label
                    class="usa-label text-normal"
                    data-testid="label"
                    for="basics.cmmiGroups"
                  >
                    CMMI Group
                  </label>
                  <p
                    class="text-base margin-bottom-1 margin-top-1"
                  >
                    You only need to select the CMMI group if CMMI is selected as the main CMS component.
                  </p>
                  <div
                    class="mint-checkbox usa-checkbox easy-checkbox--disabled"
                  >
                    <input
                      class="usa-checkbox__input"
                      disabled=""
                      id="new-plan-cmmiGroup-PATIENT_CARE_MODELS_GROUP"
                      name="basics.cmmiGroups"
                      type="checkbox"
                      value="PATIENT_CARE_MODELS_GROUP"
                    />
                    <label
                      class="usa-checkbox__label"
                      for="new-plan-cmmiGroup-PATIENT_CARE_MODELS_GROUP"
                    >
                      Patient Care Models Group (PCMG)
                    </label>
                    <p
                      class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs text-gray-30"
                    />
                  </div>
                  <div
                    class="mint-checkbox usa-checkbox easy-checkbox--disabled"
                  >
                    <input
                      class="usa-checkbox__input"
                      disabled=""
                      id="new-plan-cmmiGroup-POLICY_AND_PROGRAMS_GROUP"
                      name="basics.cmmiGroups"
                      type="checkbox"
                      value="POLICY_AND_PROGRAMS_GROUP"
                    />
                    <label
                      class="usa-checkbox__label"
                      for="new-plan-cmmiGroup-POLICY_AND_PROGRAMS_GROUP"
                    >
                      Policy and Programs Group (PPG)
                    </label>
                    <p
                      class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs text-gray-30"
                    />
                  </div>
                  <div
                    class="mint-checkbox usa-checkbox easy-checkbox--disabled"
                  >
                    <input
                      class="usa-checkbox__input"
                      disabled=""
                      id="new-plan-cmmiGroup-SEAMLESS_CARE_MODELS_GROUP"
                      name="basics.cmmiGroups"
                      type="checkbox"
                      value="SEAMLESS_CARE_MODELS_GROUP"
                    />
                    <label
                      class="usa-checkbox__label"
                      for="new-plan-cmmiGroup-SEAMLESS_CARE_MODELS_GROUP"
                    >
                      Seamless Care Models Group (SCMG)
                    </label>
                    <p
                      class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs text-gray-30"
                    />
                  </div>
                  <div
                    class="mint-checkbox usa-checkbox easy-checkbox--disabled"
                  >
                    <input
                      class="usa-checkbox__input"
                      disabled=""
                      id="new-plan-cmmiGroup-STATE_AND_POPULATION_HEALTH_GROUP"
                      name="basics.cmmiGroups"
                      type="checkbox"
                      value="STATE_AND_POPULATION_HEALTH_GROUP"
                    />
                    <label
                      class="usa-checkbox__label"
                      for="new-plan-cmmiGroup-STATE_AND_POPULATION_HEALTH_GROUP"
                    >
                      State and Population Health Group (SPHG)
                    </label>
                    <p
                      class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs text-gray-30"
                    />
                  </div>
                  <div
                    class="mint-checkbox usa-checkbox easy-checkbox--disabled"
                  >
                    <input
                      class="usa-checkbox__input"
                      disabled=""
                      id="new-plan-cmmiGroup-TBD"
                      name="basics.cmmiGroups"
                      type="checkbox"
                      value="TBD"
                    />
                    <label
                      class="usa-checkbox__label"
                      for="new-plan-cmmiGroup-TBD"
                    >
                      To be determined
                    </label>
                    <p
                      class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs text-gray-30"
                    />
                  </div>
                </div>
                <div
                  class="margin-top-6 margin-bottom-3"
                >
                  <button
                    class="usa-button"
                    data-testid="button"
                    type="submit"
                  >
                    Next
                  </button>
                </div>
                <button
                  class="usa-button usa-button usa-button--unstyled"
                  data-testid="button"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="usa-icon margin-right-1"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                    />
                  </svg>
                  Save and return to task list
                </button>
              </form>
            </div>
            <div
              class="desktop:grid-col-6"
              data-testid="grid"
            >
              <div
                class="usa-summary-box margin-top-6"
                data-testid="summary-box--previous-name"
              >
                <div
                  class="usa-summary-box__body"
                >
                  <h3
                    class="usa-summary-box__heading"
                  />
                  <div
                    class="usa-summary-box__text"
                  >
                    <p
                      class="margin-y-0 text-bold"
                    >
                      Previous model names
                    </p>
                    <ul
                      class="margin-top-1 margin-bottom-0 padding-left-2"
                    >
                      <li>
                        First Name
                      </li>
                      <li>
                        Second Name
                      </li>
                      <li>
                        Third Name
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mint-page-number margin-bottom-10"
        >
          <span
            class="mint-page-number__page-num"
          >
            Page 1 of 3
          </span>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
